_RFCs/0002-scaffolding-rfc.md

```diff
@@ -0,0 +1,57 @@
+- Change Name: scaffolding-rfc
+- Start Date: 2017-11-30
+- RFC PR: (leave this empty)
+- Core-Plans Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+This RFC is meant to clarify the purpose and general functionality expected of Habitat scaffoldings - including (but not limited to) the node scaffolding and ruby scaffolding.
+
+# Motivation
+[motivation]: #motivation
+
+There has been some healthy debate in the Habitat community around:
+* How much functionality (and how customizable is should be) should be included in a scaffolding
+* What the intended audience is for Habitat scaffoldings (i.e. beginner or experienced Habitat users)
+* Whether a scaffolding should include a runtime for languages/frameworks that can be compiled into static assets.
+
+The expected outcome of this RFC is adoption of community expectations and standards for scaffoldings within the core-plans repository.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Core Habitat scaffoldings should provide the quickest and most delightful path to packaging and running an application in particular languages. The core audience for scaffoldings are new users of Habitat - who are still learning the Habitat concepts, terminology, and technology components. A first time Habitat user should be able to use a scaffolding to package and run a simple application over a lunch break.
+
+That said - an experienced user of Habitat should be able to use and build upon a scaffolding to package and run more complex applications. Scaffoldings should allow customization through overriding callbacks, environmental variables, plan settings, helper functions, and more. However, this kind of customization should not be required for simple applications - it should be an option, not a requirement.
+
+Additionally, scaffoldings should not require the use of alpha features - features should be out of alpha and beta to be required by scaffoldings. Although users are welcome to experiment with combining alpha features and scaffolding, they are not officially supported by the core-plans maintainers.
+
+# Reference-level explanation
```

Comment:
I'm trying to think about what I might be looking for out of this RFC if I were to start a new scaffolding.  Currently, this section seems to set down 3 requirements:

- A scaffolding must include an application runtime
- A scaffolding cannot depend on alpha or beta features
- A scaffolding must be easy to use (able to package within 1 hour with minimal experience)

The third one is a bit more subjective from an implementors perspective, but does feel useful to include.

I'm wondering: are there any other requirements?  What do the existing scaffoldings do that we've found successful enough to standardize.  

Put another way, this RFC focuses on the recently contentious behavior, but since this is an RFC that people might look to when implementing scaffoldings, it might be good to also list any other criteria that we generally agree on.